src/ApiService/ApiService/Log.cs

```diff
@@ -121,77 +121,80 @@ public interface ILogTracer
 
 public class LogTracer : ILogTracer
 {
+    private string? GetCaller()
+    {
+        return new StackTrace()?.GetFrame(2)?.GetMethod()?.DeclaringType?.FullName;
+    }
 
-    private List<ILog> loggers;
+    private List<ILog> _loggers;
 
-    private IDictionary<string, string> tags = new Dictionary<string, string>();
-    private Guid correlationId;
+    public Guid CorrelationId { get; }
+    public IDictionary<string, string> Tags { get; }
 
     public LogTracer(Guid correlationId, List<ILog> loggers)
     {
-        this.correlationId = correlationId;
-        this.loggers = loggers;
+        CorrelationId = correlationId;
+        Tags = new Dictionary<string, string>();
+        _loggers = loggers;
     }
 
-    public IDictionary<string, string> Tags => tags;
-
     public void Info(string message)
     {
-        var caller = new StackTrace()?.GetFrame(1)?.GetMethod()?.Name;
-        foreach (var logger in loggers)
+        var caller = GetCaller();
+        foreach (var logger in _loggers)
         {
-            logger.Log(correlationId, message, SeverityLevel.Information, Tags, caller);
+            logger.Log(CorrelationId, message, SeverityLevel.Information, Tags, caller);
```

Comment:
These calls modify the tags of this instance. It means the logger is not thread safe, is that intentional ?

Reply:
I'll change to a record , so adding  new tags will create  a new logger.

Here logger is just a bag of properties, and ideally you just clone it per thread.